client/state/ui/first-view/selectors.js

```diff
@@ -49,3 +58,18 @@ export function shouldViewBeVisible( state ) {
 		switchedFromDifferentSection( state ) &&
 		! isSectionLoading( state );
 }
+
+function isLikelyInSafariPrivateMode() {
+	// Safari doesn't allow access to Local Storage when in Private Mode;
+	// We will use this check as a proxy to determine whether or not Safari
+	// is in Private Mode
+	let isAbleToAccessLocalStorage = true;
+
+	try {
+		localStorage.checkIfFirstViewPreferencesCanBeCached = null;
```

Comment:
`localStorage` is on the `window` object, so this would break SSR constraints. Do we actually want to test for localStorage here? We use `localForage`, I wonder if there's an established way to check elsewhere in Calypso.

Also, putting this in the selector worries me.